### PR TITLE
[VENDOR] Update dependencies CAT-317

### PR DESCRIPTION
**Pull request by norvin-yecla-lrn**
* Update dependencies
* Make tests work in JUnit 5
* Fix wrong argument in `test` command inside Makefile
* Update compile target and source to Java 1.7

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -13,23 +13,24 @@
...
       <groupId>commons-codec</groupId>
       <artifactId>commons-codec</artifactId>
-      <version>1.11</version>
+      <version>1.13</version>
     </dependency>
     <dependency>
       <groupId>org.apache.commons</groupId>
       <artifactId>commons-lang3</artifactId>
-      <version>3.5</version>
+      <version>3.8.1</version>
```
Comment (norvin-yecla-lrn) [inline pom.xml]: As of the time this PR was filed, there exists a `3.9` version, but it compiles to Java 1.8. `3.8.1` on the other hand compiles to Java 1.7 which is the oldest version we support.

File: src/test/java/learnositysdk/request/DataApiIT.java [hunk 2]
```diff
@@ -8,10 +8,14 @@
 import org.json.JSONArray;
 import org.apache.http.client.config.RequestConfig;
 
-import junit.framework.TestCase;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
```
Comment (flakeparadigm) [inline src/test/java/learnositysdk/request/DataApiIT.java]: This import (`Disabled`) is unused.

File: src/test/java/learnositysdk/request/DataApiIT.java [hunk 3]
```diff
@@ -8,10 +8,14 @@
...
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
```
Comment (flakeparadigm) [inline src/test/java/learnositysdk/request/DataApiIT.java]: This import (`assertEquals`) is unused.

File: src/test/java/learnositysdk/request/DataApiIT.java [hunk 4]
```diff
@@ -196,9 +207,7 @@ public void assertConsistentResponseRemote(Remote responseRemote, JSONObject res
 	private void assertConsistentResponseCodeStatus(int statusCode, boolean status)
 		throws org.json.JSONException
 	{
-		assertTrue("Inconsistent HTTP status cand and meta status",
-				(statusCode == 200 && status)
-				|| (statusCode != 200 && !status));
+		assertTrue((statusCode == 200 && status) || (statusCode != 200 && !status), "Inconsistent HTTP status cand and meta status");
```
Comment (flakeparadigm) [inline src/test/java/learnositysdk/request/DataApiIT.java]: It's probably good to keep this split across multiple lines to improve readability like:

```java
assertTrue(
    (statusCode == 200 && status) || (statusCode != 200 && !status),
    "Inconsistent HTTP status cand and meta status"
);
```

**Pull request merged**